sqs-queue/main.tf

```diff
@@ -0,0 +1,26 @@
+resource "aws_sqs_queue" "sqs_queue" {
+  name                       = var.queue_name
+  delay_seconds              = var.delay_seconds
+  max_message_size           = var.max_message_size
+  message_retention_seconds  = var.message_retention_seconds
+  receive_wait_time_seconds  = var.receive_wait_time_seconds
+  visibility_timeout_seconds = var.visibility_timeout_seconds
+  fifo_queue                 = var.fifo_queue
+  tags                       = var.tags
+  redrive_policy = var.enable_dlq ? jsonencode({
+    deadLetterTargetArn = aws_sqs_queue.sqs_queue_dlq[0].arn
+    maxReceiveCount     = var.max_receive_count
+  }) : null
+}
+
+resource "aws_sqs_queue" "sqs_queue_dlq" {
+  count                      = var.enable_dlq ? 1 : 0
+  name                       = "${var.queue_name}-DLQ"
+  delay_seconds              = var.delay_seconds
+  max_message_size           = var.max_message_size
+  receive_wait_time_seconds  = var.receive_wait_time_seconds
+  visibility_timeout_seconds = var.visibility_timeout_seconds
+  fifo_queue                 = var.fifo_queue
+  message_retention_seconds  = var.message_retention_seconds * 10
```

Comment:
Hvorfor ganger du med 10 her?

Reply:
Godt spørsmål 😄 jeg bare kopierte over det eksisterende oppsettet i sns-topic-subscription. @hcalsos Vet du noe om dette?

Reply:
Eller, det er jo for så vidt en fin måte å skrive 10 dager på :smile:

Reply:
Kan sikkert gange det med 7 i stedet ? så blir meldingen liggende en uke i stedet for 10 dager :)

Reply:
tenker uansett at vår default verdi på 1 dag, er for lite for dlq'ene i hvertfall.

Reply:
Kanskje sette den til 0 ? Da blir den vel tatt vare på til evig tid ?

Reply:
Men er det ikke ryddigere å gange med 7 når vi setter variabelen, og ikke gjøre gangingen i ressursen?

Reply:
Kunne kanskje hatt et parameter for dlq-retention med default på 0, som man kan overstyre hvis man vil?

Reply:
Nå har vi satt 1 dag som default for sqs køer og 10 dager som default for dlq. Det er relativt fornuftige verdier.